src/xcode/ENA/ENA/Source/Services/PPAnalyticsCollector/PPAnalyticsCollector.swift

```diff
@@ -124,67 +130,87 @@ enum PPAnalyticsCollector {
 
 	private static func logRiskExposureMetadata(_ riskExposureMetadata: PPARiskExposureMetadata) {
 		switch riskExposureMetadata {
-		case let .create(metaData):
-			store?.currentRiskExposureMetadata = metaData
-		case let .updateRiskExposureMetadata(riskCalculationResult):
-			Analytics.updateRiskExposureMetadata(riskCalculationResult)
+		case .update:
+			Analytics.gatherRisksAndUpdateMetadata()
+		}
+	}
+	
+	private static func gatherRisksAndUpdateMetadata() {
+		if let enfRiskCalculationResult = store?.enfRiskCalculationResult {
+			updateENFRiskExposureMetadata(enfRiskCalculationResult)
+		}
+		if let checkinRiskCalculationResult = store?.checkinRiskCalculationResult {
+			updateCheckinRiskExposureMetadata(checkinRiskCalculationResult)
 		}
 	}
 
-	private static func updateRiskExposureMetadata(_ enfRiskCalculationResult: ENFRiskCalculationResult) {
+	private static func updateENFRiskExposureMetadata(_ enfRiskCalculationResult: ENFRiskCalculationResult) {
 		let riskLevel = enfRiskCalculationResult.riskLevel
+		let previousRiskLevel = store?.previousENFRiskExposureMetadata?.riskLevel
+		let mostRecentDateWithCurrentRiskLevel = enfRiskCalculationResult.mostRecentDateWithCurrentRiskLevel
+		let previousMostRecentDateWithCurrentRiskLevel = store?.previousENFRiskExposureMetadata?.mostRecentDateAtRiskLevel
 		let riskLevelChangedComparedToPreviousSubmission: Bool
 		let dateChangedComparedToPreviousSubmission: Bool
 
-		// if there is a risk level value stored for previous submission
-		if store?.previousRiskExposureMetadata?.riskLevel != nil {
-			if riskLevel !=
-				store?.previousRiskExposureMetadata?.riskLevel {
-				// if there is a change in risk level
-				riskLevelChangedComparedToPreviousSubmission = true
-			} else {
-				// if there is no change in risk level
-				riskLevelChangedComparedToPreviousSubmission = false
-			}
-		} else {
-			// for the first time, the field is set to false
-			riskLevelChangedComparedToPreviousSubmission = false
+		// If risk level is nil, set to false. Otherwise, set it when it changed compared to previous submission.
+		riskLevelChangedComparedToPreviousSubmission = riskLevel != previousRiskLevel && previousRiskLevel != nil
```

Comment:
This refactoring needs rigorous testing. I hope we have done enough developer testing before merging into release.

Reply:
I think we including you wrote these tests. So I am optimistic that this will be correct.